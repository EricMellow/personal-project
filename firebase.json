{
  "hosting": {
    "public": "/",
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**",
      "keys.js"
    ]
  }
}
