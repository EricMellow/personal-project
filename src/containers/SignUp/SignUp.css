.sign-up {
background: linear-gradient(to top, #0b3080 0%,#1a66bf 100%); 
height: 100vh;
display: flex;
flex-direction: column;
justify-content: center;
align-items: center;
}

.input {
  background: none;
  height: 3rem;
  border: 2px solid #ecb429;
  width: 30rem;
  color: #ecb429;
  padding-left: 10px;
  margin: 20px;
  font-size: 2rem;
  border-radius: 5px;
}

.sign-up-btn {
  height: 3rem;
  width: 30rem;
  border-radius: 5px;
  font-size: 2rem;
  color: #ecb429;
  background: none;
  border: 2px solid #ecb429;
  margin: 20px;
}

*, *:before, *:after {
  box-sizing: inherit;
}