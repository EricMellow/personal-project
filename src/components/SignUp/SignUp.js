import React, { Component } from "react";
import './SignUp.css';
import { auth } from '../../firebase';
import { withRouter } from 'react-router-dom';

export class SignUp extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      password: '',
      confirmPassword: '',
      error: null
    };
  }

  handleInput = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  resetState = () => {
    this.setState({
      username: '',
      email: '',
      password: '',
      confirmPassword: '',
      error: null
    });
  }

  storeData = async (event) => {
    event.preventDefault();
    try {
      await auth.doCreateUserWithEmailAndPassword(this.state.email, this.state.password);
      this.resetState();
      this.props.history.push('/');
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  render() {
    const {
      username,
      email,
      password,
      confirmPassword,
      error
    } = this.state;

    const invalidUserInfo =
      password !== confirmPassword ||
      password === '' ||
      email === '' ||
      username === '';

    return (
      <form onSubmit={event => this.storeData(event)}>
        <input
          placeholder="Full Name"
          name="username"
          value={username}
          onChange={event => this.handleInput(event)}
          type="text"
        />
        <input
          placeholder="Email Address"
          name="email"
          value={email}
          onChange={event => this.handleInput(event)}
          type="text"
        />
        <input
          placeholder="Password"
          name="password"
          value={password}
          onChange={event => this.handleInput(event)}
          type="password"
        />
        <input
          placeholder="Confirm Password"
          name="confirmPassword"
          value={confirmPassword}
          onChange={event => this.handleInput(event)}
          type="password"
        />
        <button type="submit" disabled={invalidUserInfo}>
          Sign Up
        </button>

        {error && <p>{error}</p>}
      </form>
    );
  }
}

export default SignUp;