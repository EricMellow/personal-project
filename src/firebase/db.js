import { db } from './firebase';

// User API

export const doCreateUser = (id, username, email, zipcode) =>
  db.ref(`users/${id}`).set({
    username,
    email,
    zipcode
  });

export const doCreateActivity = (address, duration, lat, lng, type) => 
  db.ref('actions/').push({
    address,
    type,
    duration,
    lat,
    lng
  })

export const doGetAllActivities = () => 
  db.ref('actions/').on('value', snapshot => snapshot.val())

// export const onceGetUsers = () =>
//   db.ref('users').once('value');