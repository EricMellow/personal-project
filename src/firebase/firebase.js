import { firebaseKey } from '../keys';
import * as firebase from 'firebase';
const key = firebaseKey;
// Initialize Firebase
var config = {
  apiKey: key,
  authDomain: "personal-project-1527122816740.firebaseapp.com",
  databaseURL: "https://personal-project-1527122816740.firebaseio.com",
  projectId: "personal-project-1527122816740",
  storageBucket: "",
  messagingSenderId: "182263229649"
};

if (!firebase.apps.length) {
  firebase.initializeApp(config);
}

const db = firebase.database();
const auth = firebase.auth();

export { db, auth };