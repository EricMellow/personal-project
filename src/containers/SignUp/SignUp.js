import React, { Component } from "react";
import './SignUp.css';
import { auth, db } from '../../firebase';
import { connect } from 'react-redux';
import { authenticateUser } from "../../actions/authenticateUser";
import { addUserId } from "../../actions/userIdActions";

export class SignUp extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      password: '',
      confirmPassword: '',
      zipcode: undefined,
      error: null
    };
  }

  handleInput = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  resetState = () => {
    this.setState({
      username: '',
      email: '',
      password: '',
      confirmPassword: '',
      zipcode: undefined,
      error: null
    });
  }

  storeData = async (event) => {
    const {
      email,
      password,
      username,
      zipcode
    } = this.state;

    event.preventDefault();
    try {
      const authUser = await auth.doCreateUserWithEmailAndPassword(email, password);
      await db.doCreateUser(authUser.user.uid, username, email, zipcode);
      this.resetState();
      this.props.storeUserId(authUser.user.uid);
      this.props.authenticate();
      this.props.history.push('/');
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  render() {
    const {
      username,
      email,
      password,
      confirmPassword,
      zipcode,
      error
    } = this.state;

    const invalidUserInfo =
      password !== confirmPassword ||
      password === '' ||
      email === '' ||
      username === '' ||
      zipcode === null;

    return (
      <form className="sign-up" onSubmit={event => this.storeData(event)}>
        <input
          placeholder="Username"
          name="username"
          className="input name-input"
          value={username}
          onChange={event => this.handleInput(event)}
          type="text"
        />
        <input
          placeholder="Zip Code"
          name="zipcode"
          className="input zip-input"
          value={zipcode}
          onChange={event => this.handleInput(event)}
          type="number"
        />
        <input
          placeholder="Email Address"
          name="email"
          className="input email-input"
          value={email}
          onChange={event => this.handleInput(event)}
          type="text"
        />
        <input
          placeholder="Password"
          name="password"
          className="input password-input"
          value={password}
          onChange={event => this.handleInput(event)}
          type="password"
        />
        <input
          placeholder="Confirm Password"
          name="confirmPassword"
          className="input confirm-input"
          value={confirmPassword}
          onChange={event => this.handleInput(event)}
          type="password"
        />
        <button className="sign-up-btn" type="submit" disabled={invalidUserInfo}>
          Sign Up
        </button>

        {error && <p>{error}</p>}
      </form>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  authenticate: () => dispatch(authenticateUser()),
  storeUserId: (userId) => dispatch(addUserId(userId))
});


export default connect(null, mapDispatchToProps)(SignUp);