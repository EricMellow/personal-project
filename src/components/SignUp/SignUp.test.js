import SignUp from "./SignUp";
import { shallow } from "enzyme";
import React from "react";
import { auth } from "../../firebase";

describe('SignUp', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SignUp />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleInput', () => {

    it('should set the correct key/value pair in state with the incomming event data', () => {
      const wrapper = shallow(<SignUp />);
      const mockEvent = {
        target: {
          name: 'password',
          value: 'trustnoone'
        }
      };
      const initialExpected = {
        username: '',
        email: '',
        password: '',
        confirmPassword: '',
        error: null
      };
      const finalExpected = {
        username: '',
        email: '',
        password: 'trustnoone',
        confirmPassword: '',
        error: null
      };

      expect(wrapper.state()).toEqual(initialExpected);
      wrapper.instance().handleInput(mockEvent);
      expect(wrapper.state()).toEqual(finalExpected);
    });
  });

  describe('resetState', () => {

    it('should reset the state to the default settings', () => {
      const wrapper = shallow(<SignUp />);
      const initialState = {
        username: 'test',
        email: 'test@test.com',
        password: 'password',
        confirmPassword: 'password',
        error: null
      };
      wrapper.setState(initialState);
      const expected = {
        username: '',
        email: '',
        password: '',
        confirmPassword: '',
        error: null
      };

      expect(wrapper.state()).toEqual(initialState);
      wrapper.instance().resetState();
      expect(wrapper.state()).toEqual(expected);
    });
  });

  describe('storeData', () => {
    let wrapper;
    let mockHistory;
    let mockEvent;

    beforeEach(() => {
      mockEvent = { preventDefault: jest.fn() }
      mockHistory = { push: jest.fn() };
      wrapper = shallow(<SignUp history={mockHistory}/>);
      auth.doCreateUserWithEmailAndPassword = jest.fn().mockImplementation(() => Promise.resolve({}));
    });

    it('should call doCreateUserWithEmailAndPassword with the correct arguments', () => {
      wrapper.setState({
        email: 'test@test.com',
        password: 'trustnoone'
      });

      wrapper.instance().storeData(mockEvent);
      expect(auth.doCreateUserWithEmailAndPassword).toHaveBeenCalledWith('test@test.com', 'trustnoone')
    });

    it('should reset the state to the default parameters', async () => {
      const initialState = {
        username: 'test',
        email: 'test@test.com',
        password: 'password',
        confirmPassword: 'password',
        error: null
      };
      wrapper.setState(initialState);
      const expected = {
        username: '',
        email: '',
        password: '',
        confirmPassword: '',
        error: null
      };

      expect(wrapper.state()).toEqual(initialState);
      await wrapper.instance().storeData(mockEvent);
      expect(wrapper.state()).toEqual(expected);
    });

    // it('should set the error key in state with a value of the error message if there is an error', () => {
    //   auth.doCreateUserWithEmailAndPassword = jest.fn().mockImplementation(() => Promise.reject(new Error({message: 'oops!'})));
    //   const expected = 'oops!';
      
    //   wrapper.instance().storeData(mockEvent);
    //   expect(wrapper.state('error')).toEqual(expected);

    // });

  });

  

});