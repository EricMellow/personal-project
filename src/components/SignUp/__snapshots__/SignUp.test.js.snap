// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="input name-input"
          name="username"
          onChange={[Function]}
          placeholder="Full Name"
          type="text"
          value=""
/>,
        <input
          className="input email-input"
          name="email"
          onChange={[Function]}
          placeholder="Email Address"
          type="text"
          value=""
/>,
        <input
          className="input password-input"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
/>,
        <input
          className="input confirm-input"
          name="confirmPassword"
          onChange={[Function]}
          placeholder="Confirm Password"
          type="password"
          value=""
/>,
        <button
          disabled={true}
          type="submit"
>
          Sign Up
</button>,
        null,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input name-input",
          "name": "username",
          "onChange": [Function],
          "placeholder": "Full Name",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input email-input",
          "name": "email",
          "onChange": [Function],
          "placeholder": "Email Address",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input password-input",
          "name": "password",
          "onChange": [Function],
          "placeholder": "Password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input confirm-input",
          "name": "confirmPassword",
          "onChange": [Function],
          "placeholder": "Confirm Password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
          "disabled": true,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "button",
      },
      null,
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="input name-input"
            name="username"
            onChange={[Function]}
            placeholder="Full Name"
            type="text"
            value=""
/>,
          <input
            className="input email-input"
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            type="text"
            value=""
/>,
          <input
            className="input password-input"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
/>,
          <input
            className="input confirm-input"
            name="confirmPassword"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
            value=""
/>,
          <button
            disabled={true}
            type="submit"
>
            Sign Up
</button>,
          null,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input name-input",
            "name": "username",
            "onChange": [Function],
            "placeholder": "Full Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input email-input",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email Address",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input password-input",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input confirm-input",
            "name": "confirmPassword",
            "onChange": [Function],
            "placeholder": "Confirm Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "disabled": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "button",
        },
        null,
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
